crates/matrix-sdk-ui/src/timeline/tests/polls.rs

```diff
@@ -0,0 +1,247 @@
+        self.events().await.first().unwrap().clone().poll_state()
+    }
+
+    async fn send_poll_start(&self, sender: &UserId, content: UnstablePollStartContentBlock) {
+        let event_content = AnyMessageLikeEventContent::UnstablePollStart(
+            UnstablePollStartEventContent::new(content),
+        );
+        self.handle_live_message_event(sender, event_content).await;
+    }
+
+    async fn send_poll_start_with_id(
+        &self,
+        sender: &UserId,
+        event_id: &EventId,
+        content: UnstablePollStartContentBlock,
+    ) {
+        let event_content = AnyMessageLikeEventContent::UnstablePollStart(
+            UnstablePollStartEventContent::new(content),
+        );
+        let event = self.make_message_event_with_id(sender, event_content, event_id.to_owned());
+        let raw = Raw::new(&event).unwrap().cast();
+        self.handle_live_event(raw).await;
+    }
+
+    async fn send_poll_response(&self, sender: &UserId, answers: Vec<&str>, poll_id: &EventId) {
+        let event_content = AnyMessageLikeEventContent::UnstablePollResponse(
+            UnstablePollResponseEventContent::new(
+                answers.into_iter().map(|i| i.to_owned()).collect(),
+                poll_id.to_owned(),
+            ),
+        );
+        self.handle_live_message_event(sender, event_content).await
+    }
+
+    async fn send_poll_end(&self, sender: &UserId, text: &str, poll_id: &EventId) {
+        let event_content = AnyMessageLikeEventContent::UnstablePollEnd(
+            UnstablePollEndEventContent::new(text, poll_id.to_owned()),
+        );
+        self.handle_live_message_event(sender, event_content).await
+    }
+
+    async fn send_poll_edit(
+        &self,
+        sender: &UserId,
+        original_id: &EventId,
+        content: UnstablePollStartContentBlock,
+    ) {
+        let mut content = UnstablePollStartEventContent::new(content);
+        content.relates_to = Some(Relation::Replacement(Replacement::new(
+            original_id.to_owned(),
+            content.clone().into(),
+        )));
+        let event_content = AnyMessageLikeEventContent::UnstablePollStart(content);
+        self.handle_live_message_event(sender, event_content).await
+    }
+}
+
+impl EventTimelineItem {
+    fn poll_state(self) -> PollState {
+        match self.content() {
+            TimelineItemContent::Poll(poll_state) => poll_state.clone(),
+            _ => panic!("Not a poll state"),
+        }
+    }
+}
+
+fn assert_poll_start_eq(a: &UnstablePollStartContentBlock, b: &UnstablePollStartContentBlock) {
+    assert_eq!(a.question.text, b.question.text);
+    assert_eq!(a.kind, b.kind);
+    assert_eq!(a.max_selections, b.max_selections);
+    assert_eq!(a.answers.len(), a.answers.len());
+    assert_eq!(a.answers[0].id, a.answers[0].id);
+    assert_eq!(a.answers[0].text, a.answers[0].text);
+    assert_eq!(a.answers[1].id, a.answers[1].id);
+    assert_eq!(a.answers[1].text, a.answers[1].text);
```

Comment:
Not sure if you've seen the clippy warnings about this from CI?

```suggestion
    assert_eq!(a.answers[0].id, b.answers[0].id);
    assert_eq!(a.answers[0].text, b.answers[0].text);
    assert_eq!(a.answers[1].id, b.answers[1].id);
    assert_eq!(a.answers[1].text, b.answers[1].text);
```

Although why not just loop over all the answers?